src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs

```diff
@@ -18,11 +18,12 @@ public HttpClientHandler_AltSvc_Test(ITestOutputHelper output) : base(output)
         /// <summary>
         /// HTTP/3 tests by default use prenegotiated HTTP/3. To test Alt-Svc upgrades, that must be disabled.
         /// </summary>
-        protected override HttpClient CreateHttpClient()
+        private HttpClient CreateHttpClient(Version version)
         {
-            HttpClientHandler handler = CreateHttpClientHandler(HttpVersion.Version30);
-
-            return CreateHttpClient(handler);
+            var client = base.CreateHttpClient();
```

Comment:
Is it ok that version used in `base.CreateHttpClientHandler()` inside `base.CreateHttpClient()` will not be the same that is later set here to `client.DefaultRequestVersion`? Maybe it will be better to follow previous pattern to create handler and supply it to new client?

Reply:
I not 100% sure I understand what you mean. Could you share a snippet?

Reply:
If you call `CreateHttpClient()` it will call `CreateHttpClientHandler()` without parameters
https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs#L34
Which in its turn will pass `UseVersion`
https://github.com/dotnet/runtime/blob/a822d3909b90139b642b143e889505ab7157b10c/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs#L60
So the version set to `HttpClient` and the version used for handler creating will be different.
What happens with `UseVersion` in `CreateHttpClientHandler` might not be *that* important at the moment, but I have a feeling it might lead to bugs in the future in case `CreateHttpClientHandler` would be rewritten some day.

Reply:
`UseVersion` is overidden to H/2 and H/3 in the concrete class, e.g.:
https://github.com/ManickaP/runtime/blob/a07bc51c03add6e56bb5e36610f776584f52f8e6/src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs#L3161-L3161

> So the version set to HttpClient and the version used for handler creating will be different.

That's what I'm aiming for here though :smile: I need the client handler to be created with appropriate `QuicImplementationProvider` but also to have the request with different version than what the handler was created so that `VersionHttpClientHandler` does not overrides the `VersionPolicy` to exact and we can actually test the AltSvc.

It doesn't matter here if I use `CreateHttpClient` or `CreateHttpClientHandler`, I'd still need to set up `DefaultVersionPolicy` and `DefaultRequestVersion` on the client.